src/client.rs

```diff
@@ -612,16 +612,16 @@ impl RequestBuilder {
     ///
     /// let client = reqwest::Client::new()?;
     /// let res = client.post("http://httpbin.org")
-    ///     .json(&map)
+    ///     .json(&map)?
     ///     .send()?;
     /// # Ok(())
     /// # }
     /// ```
```

Comment:
Now that the `json` method is fallible we should add an Errors section to the docs. Here's the relevant [API guideline](https://github.com/brson/rust-api-guidelines#c-error-doc) with some inspiration.